lib/atomic/activities/session.ex

```
@@ -48,6 +48,8 @@ defmodule Atomic.Activities.Session do
     |> cast(attrs, @required_fields ++ @optional_fields)
     |> validate_required(@required_fields)
     |> validate_location()
+    |> validate_dates()
+    |> validate_entries_number()
```

This check already exists. It's literally on the line below 😅

---

I didn't notice but it wasn't working anyway. I still could create sessions ending before the starting date and with a maximum number of participants lower than the minimum

---

Are you sure about the second one? Because there is a constraint at the database level - refer to https://github.com/cesium/atomic/blob/6e13421c81cf2cf3b8a939e917b9ced878618bef/priv/repo/migrations/20221023200102_create_sessions.exs#L22

---

How have you tested it? On a forms? If so, maybe the error is not being handled

---

I tested on a forms. The way I did it now prevents it from happening and shows a error message to the user, better UX overall

---

Nice 🤩